### PR TITLE
This is the commit for my new branch with SingleProduct in it.

### PR DESCRIPTION
**Pull request by richardedwardbrown3**
"Product / Product:id "

### CODE REVIEW

File: tests/api.spec.js [hunk 1]
```diff
@@ -2,10 +2,10 @@
 const axios = require('axios');
 require('dotenv').config();
 const bcrypt = require('bcrypt');
-const jwt = require('jsonwebtoken');
+// const jwt = require('jsonwebtoken');
 const { SERVER_ADDRESS = 'http://localhost:', PORT = 3000 } = process.env;
 const API_URL = process.env.API_URL || SERVER_ADDRESS + PORT;
-const { JWT_SECRET = 'neverchane' } = process.env;
+// const { JWT_SECRET = 'neverchane' } = process.env;
```
Comment (Felix-Cadiz) [inline tests/api.spec.js]: It is okay to import JWT_SECRET into this file, but its value should be stored in the .env file.

File: tests/api.spec.js [hunk 2]
```diff
@@ -32,7 +32,16 @@ describe('API', () => {
     expect(typeof res.data.message).toEqual('string');
   });
   describe('Users', () => {
-    let newUser = { username: 'Ricky', password: 'BrownBrown' };
+    const newUser = { 
+      id:  1,
```
Comment (Felix-Cadiz) [inline tests/api.spec.js]: Because id's are serialized, the defined id may not be needed.

File: tests/api.spec.js [hunk 3]
```diff
@@ -108,10 +117,10 @@ describe('API', () => {
       });
     });
   describe('products', () => {
-    const productToCreate = {name: "Test_Wine_Or_Cheese", description: "Test_Description", price: "$Test", imgURL: "imageUrl", inStock: true, category: "Test category"};
+    const productToCreate = {id: 0, name: "Test_Wine_Or_Cheese", description: "Test_Description", price: "2", imgURL: "imageUrl", inStock: true, category: "Test category"};
```
Comment (Felix-Cadiz) [inline tests/api.spec.js]: Because id's are serialized, the defined id may not be needed.

File: tests/api.spec.js [hunk 4]
```diff
@@ -108,10 +117,10 @@ describe('API', () => {
...
     describe('GET /', () => {
-      it('Just returns a list of all products in the database', async () => {
-        const product = { name: 'Grenache', inStock: true };
+      it('Returns a list of all products in the database', async () => {
+        const product = { id: 1, name: 'Grenache', inStock: true };
```
Comment (Felix-Cadiz) [inline tests/api.spec.js]: Because id's are serialized, the defined id may not be needed.

File: tests/db.spec.js [hunk 5]
```diff
@@ -20,31 +21,39 @@ describe('Database', () => {
...
+          imgURL: "nopelol.com",
+          username: "rickybobby",
+          password: "rickybobby", 
+          isAdmin: true, 
+          address: "yee"};
       describe('createUser({ username, password })', () => {
         beforeAll(async () => {
           userToCreateAndUpdate = await createUser(userCredentials);
-          const {rows} = await client.query(`SELECT * FROM users WHERE username = $1`, [userCredentials.username]);
+          const {rows} = await client.query(`SELECT * FROM users WHERE username = $5`, [userCredentials.username]);
```
Comment (Felix-Cadiz) [inline tests/db.spec.js]: WHERE username = $1.

The value number that is passed in is in reference to the array that follows the query.

**Review by Felix-Cadiz — COMMENTED** [review]
A couple items to fix before the merge. Great job creating the tests!

File: tests/api.spec.js (hunk 1 above)
Comment (richardedwardbrown3) [reply tests/api.spec.js]: got it

File: tests/api.spec.js (hunk 2 above)
Comment (richardedwardbrown3) [reply tests/api.spec.js]: out

File: tests/api.spec.js (hunk 3 above)
Comment (richardedwardbrown3) [reply tests/api.spec.js]: out as well; should have done a better search

File: tests/api.spec.js (hunk 4 above)
Comment (richardedwardbrown3) [reply tests/api.spec.js]: everywhere

File: tests/db.spec.js (hunk 5 above)
Comment (richardedwardbrown3) [reply tests/db.spec.js]: ok, I was worried $1 would give me .firstname not .username

**Comment by richardedwardbrown3** [comment]
thank you. fixing.

File: tests/db.spec.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
+//Test DB 2/2/22
 require('dotenv').config();
 const bcrypt = require('bcrypt');
 const SALT_COUNT = 10;
-const { rebuildDB } = require('../db/seedData');
+Co
```
Comment (Felix-Cadiz) [inline tests/db.spec.js:5]: Small typo?

**Review by Felix-Cadiz — COMMENTED** [review]
One last edit Ricky =)

**Review by Felix-Cadiz — APPROVED** [review]
LGTM!

**Review by Felix-Cadiz — APPROVED** [review]
Thanks Ricky!

**Pull request merged**